rfcs/0003-multi-blockchain-support.md

```diff
@@ -0,0 +1,534 @@
+# RFC-0003: Multi-Blockchain Support
+
+<dl>
+  <dt>Author</dt>
+  <dd>Jannis Pohlmann</dd>
+
+  <dt>RFC pull request</dt>
+  <dd><a href="https://github.com/graphprotocol/rfcs/pull/8">https://github.com/graphprotocol/rfcs/pull/8</a></dd>
+
+  <dt>Obsoletes (if applicable)</dt>
+  <dd>-</dd>
+
+  <dt>Date of submission</dt>
+  <dd>2019-12-20</dd>
+
+  <dt>Date of approval</dt>
+  <dd>-</dd>
+
+  <dt>Approved by</dt>
+  <dd>-</dd>
+</dl>
+
+## Contents
+
+<!-- toc -->
+
+## Summary
+
+Multi-blockchain support allows subgraphs to index data from blockchains other
+than Ethereum.
+
+## Goals & Motivation
+
+The main objective of multi-blockchain support is to add support for subgraphs
+to index data from a growing number of blockchains. At the time of writing this
+RFC, only Ethereum and blockchains compatible with the Ethereum JSON-RPC API are
+supported.
+
+This feature unlocks the same use cases on other blockchains that The Graph has
+made possible on Ethereum, including indexing historical data, interacting with
+smart contracts (if supported by the respective chain), handling reorgs (if
+applicable) and triggering off of blockchain-specific features like events or
+transactions.
+
+## Urgency
+
+Since this is a big feature and one that requires a substantial refactoring of
+the existing codebase, and _may_ introduce breaking changes, it would be good to
+start working on this as soon as possible.
+
+## Terminology
```

Comment:
I wonder if we don't need an additional concept to encapsulate networks comprising many instances of individual chains that share some notion of causality via i.e. a hub, beacon chain, relay chain, etc.

Hard to say until we start looking at integrating Cosmos/ Polkadot whether the right granularity is indexing an individual chain directly or thinking in terms of the larger network.

Reply:
That's true. The current approach would work well for individual Polkadot para-chains in isolation and also for the relay-chain in isolation. But not for the "network of chains".